components/places/android/src/main/java/mozilla/appservices/places/PlacesConnection.kt

```diff
@@ -327,6 +327,18 @@ open class PlacesReaderConnection internal constructor(connHandle: Long) :
         }
     }
 
+    override fun getBookmarkURLForKeyword(keyword: String): String? {
+        val urlPtr = rustCall { error ->
+            LibPlacesFFI.INSTANCE.bookmarks_get_url_for_keyword(this.handle.get(), keyword, error)
+        }
+
+        try {
+            return urlPtr?.getString(0, "utf-8")
```

Comment:
Hmm, this is the second time we're returning null on success. There's a comment at the top of `matchUrl` that suggests breaking this pattern out into a `rustCallForOptString` helper, like `rustCallForString` but with an optional result. How do you feel about adding that to your PR?

Reply:
Sure, I will add another PR for that issue.